daemon/logger/bufferlog/bufferlog.go

```diff
@@ -0,0 +1,93 @@
+package bufferlog
+
+import (
+	"github.com/alibaba/pouch/daemon/logger"
+	rb "github.com/alibaba/pouch/pkg/ringbuffer"
+
+	"github.com/docker/go-units"
+	"github.com/sirupsen/logrus"
+)
+
+// BufferLog is uses to cache the container's logs with ringBuffer.
+type BufferLog struct {
+	ringBuffer *rb.RingBuffer
+	logger     logger.LogDriver
+}
+
+// NewBufferLog return a new BufferLog.
+func NewBufferLog(logDriver logger.LogDriver, info logger.Info) (logger.LogDriver, error) {
+	bufferSize := int64(-1)
+	//  translate the max buffer size if not nil
+	if s, ok := info.LogConfig["max-buffer-size"]; ok {
+		var err error
+		bufferSize, err = units.RAMInBytes(s)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	bl := &BufferLog{
+		logger:     logDriver,
+		ringBuffer: rb.New(int(bufferSize)),
+	}
+
+	// use a goroutine to write logs continuously with specified log driver
+	go bl.run()
+	return bl, nil
+}
+
+// Name return the log driver's name.
+func (bl *BufferLog) Name() string {
+	return bl.logger.Name()
+}
+
+// WriteLogMessage will write the LogMessage to the ringBuffer.
+func (bl *BufferLog) WriteLogMessage(msg *logger.LogMessage) error {
+	covered, err := bl.ringBuffer.Push(msg)
+	if covered {
+		logrus.Warnf("data coverage occurs: %v", msg)
+	}
+	return err
+}
+
+// Close close the ringBuffer and drain the messages.
+func (bl *BufferLog) Close() error {
+	bl.ringBuffer.Close()
+	var hasErr bool
+	for _, msg := range bl.ringBuffer.Drain() {
+		msgTemp, ok := msg.(*logger.LogMessage)
```

Comment:
I think we should not use `interface{}` here because the ringbuffer is only used by logging. So if we do, there is no need to assert the value each time.